lib/pages/timetable/view/timetable_page.dart

```diff
@@ -1,332 +1,379 @@
+    with TickerProviderStateMixin {
+  TimeController _timeController;
+  DateController _dateController;
+
+  @override
+  void dispose() {
+    _timeController?.dispose();
+    _dateController?.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final authProvider = Provider.of<AuthProvider>(context);
+
+    _dateController ??= DateController(
+      initialDate: DateTimeTimetable.today(),
+      visibleRange: VisibleDateRange.week(startOfWeek: DateTime.monday),
+    );
+
+    if (_timeController == null) {
+      _timeController = TimeController(
+        initialRange: TimeRange(7.hours + 55.minutes, 20.hours + 5.minutes),
+        // TODO(IoanaAlexandru): Make initialTimeRange customizable in settings
+        maxRange: TimeRange(0.hours, 24.hours),
+      );
+
+      if (authProvider.isAuthenticated && !authProvider.isAnonymous) {
+        scheduleDialog(context);
+      }
+    }
+
+    return AppScaffold(
+      title: AnimatedBuilder(
+        animation: _dateController,
+        builder: (context, child) {
+          return Text(authProvider.isAuthenticated && !authProvider.isAnonymous
+              ? _dateController.currentMonth.titleCase
+              : S.current.navigationTimetable);
+        },
+      ),
+      needsToBeAuthenticated: true,
+      leading: AppScaffoldAction(
+        icon: Icons.today_outlined,
+        onPressed: () {
+          _dateController.animateToToday(vsync: this);
+        },
+        tooltip: S.current.actionJumpToToday,
+      ),
+      actions: [
+        AppScaffoldAction(
+          icon: FeatherIcons.bookOpen,
+          tooltip: S.current.navigationClasses,
+          onPressed: () => Navigator.of(context).push(
+            MaterialPageRoute<ChangeNotifierProvider>(
+              builder: (_) => ChangeNotifierProvider.value(
+                  value: Provider.of<ClassProvider>(context),
+                  child: const ClassesPage()),
+            ),
+          ),
+        ),
+        AppScaffoldAction(
+          icon: FeatherIcons.filter,
+          tooltip: S.current.navigationFilter,
+          onPressed: () => Navigator.push(
+            context,
+            MaterialPageRoute<FilterPage>(builder: (_) => const FilterPage()),
+          ),
+        ),
+      ],
+      body: Padding(
+        padding: const EdgeInsets.all(10),
+        child: Stack(
+          children: [
+            ValueListenableBuilder<DatePageValue>(
+              valueListenable: _dateController,
+              builder: (context, value, child) {
+                final Stream<List<UniEventInstance>> eventsInRange =
+                    Provider.of<UniEventProvider>(context, listen: false)
+                        .getEventsIntersecting(
+                  Interval(
+                    // Events are preloaded for previous, current and next page
+                    DateTimeTimetable.dateFromPage(value.page.floor()) - 7.days,
```

Comment:
Ik we use 7 days for the timetable, but shouldn't this technically be `value.visibleDayCount` as well? And below too.

Reply:
So something like between floor - visibleDays and ceil + visibleDays?